### ISSUE TITLE
Why use Adam instead of AdamW as default ?

### ISSUE THREAD

**Issue by richarddwang** [PyTorchLightning/lightning-flash#250]
I found `Task` using `Adam` as default optimizer, 
but afaik both Pytorch and Tensorflow has an wrong implementation of `Adam` w.r.t weight decay,
so AdamW comes as the corrected one and soon be the default optimizer for many librarys.

**Comment by edgarriba** [comment]
thanks for the suggestion. Let us benchmark the current implementations and definitely we can adapt.

**Comment by edgarriba** [comment]
@richarddwang I did a small benchmark based on our current image classification example comparing `Adam` vs `AdamW`.

red - AdamW
orange - Adam

![image](https://user-images.githubusercontent.com/5157099/116891041-8e408c00-ac2e-11eb-97b2-5791b01ba0bd.png)

This is part of the code I used:
- num_epochs: 100
- lr: 1e-3
- backbone: "resnet18"

```python
# 2. Load the data
datamodule = ImageClassificationData.from_folders(
    train_folder="data/hymenoptera_data/train/",
    val_folder="data/hymenoptera_data/val/",
    test_folder="data/hymenoptera_data/test/",
    batch_size=8,
)

# 3. Build the model
model = ImageClassifier(
    num_classes=datamodule.num_classes,
    backbone="resnet18",
    optimizer=torch.optim.AdamW
)

# 4. Create the trainer. Run once on data
trainer = flash.Trainer(max_epochs=100, gpus=1)

# 5. Finetune the model
trainer.finetune(model, datamodule=datamodule, strategy="freeze")

# 6. Save it!
trainer.save_checkpoint("image_classification_model.pt")
```

Let me know about your thoughts.

**Comment by richarddwang** [comment]
Sorry for the late notice.

I am not a master of optimizer. But maybe you can refer to this paper "[Decoupled Weight Decay Regularization](https://arxiv.org/abs/1711.05101)" that bring out AdamW, in the paper it states

> We provide empirical evidence that our proposed modification (i) decouples the optimal choice of weight decay factor from the setting of the learning rate for both standard SGD and Adam and (ii) substantially improves Adam's generalization performance
